proposals/improved-interpolated-strings.md

```diff
@@ -112,100 +113,83 @@ Here, because `TraceLoggerParamsBuilder` has static method called `Create` with
-
-Addionally, all calls to `AppendLiteral` or `AppendFormat` must return the same type. It is not permitted to mix `bool` and `void` returning methods.
-
 The rest of this proposal will use `Append...` to refer to either of `AppendLiteral` or `AppendFormatted` in cases when both are applicable.
 
-Note that these rules do not permit extension methods for the `Append...` calls. We could consider enabling that if we choose, but this is analogous to the enumerator
-pattern, where we allow `GetEnumerator` to be an extension method, but not `Current` or `MoveNext()`.
-
-These rules _do_ permit default parameters for the `Append...` calls, which will work with things like `CallerLineNumber` or `CallerArgumentExpression` (when supported by
-the language).
-
-We have separate overload lookup rules for base elements vs interpolation holes because some builders will want to be able to understand the difference between the components
-that were interpolated and the components that were part of the base string.
+#### New attributes
 
-**Open Question**
+The compiler recognizes the `System.Runtime.CompilerServices.InterpolatedStringBuilderAttribute`:
 
-Some scenarios, like structured logging, want to be able to provide names for interpolation elements. For example, today a logging call might look like
-`Log("{name} bought {itemCount} items", name, items.Count);`. The names inside the `{}` provide important structure information for loggers that help with ensuring output
-is consistent and uniform. Some cases might be able to reuse the `:format` component of an interpolation hole for this, but many loggers already understand format specifiers
-and have existing behavior for output formatting based on this info. Is there some syntax we can use to enable putting these named specifiers in?
+```cs
+using System;
+namespace System.Runtime.CompilerServices
+{
+    [AttributeUsage(AttributeTargets.Class | AttributeTargets.Struct, AllowMultiple = false, Inherited = false)]
+    public sealed class InterpolatedStringBuilderAttribute : Attribute
+    {
+        public InterpolatedStringBuilderAttribute()
+        {
+        }
 
-Some cases may be able to get away with `CallerArgumentExpression`, provided that support does land in C# 10. But for cases that invoke a method/property, that may not be
-sufficient.
+        public bool UseLazyComputation { get; }
+    }
+}
+```
 
-#### Interpolated string builder conversion
+This attribute is used by the compiler to determine if a type is a valid interpolated string builder type.
 
-We add a new implicit conversion type: The _implicit\_string\_builder\_conversion_. An _implicit\_string\_builder\_conversion_ permits an _interpolated\_string\_expression_
-to be converted to an _applicable\_interpolated\_string\_builder\_type_. There are 2 ways that this conversion can occur:
+The compiler also recognizes the `System.Runtime.CompilerServices.InterpolatedStringBuilderAttribute`:
```

Comment:
It looks like we already said this at the beginning of the "New attributes section".